src/ast.rs

```diff
@@ -50,6 +92,6 @@ pub enum PrimitiveValue {
     Enumerated(u32),
     // Date, // not yet implemented as they dont have close enough equvalents in rust
     // Time,
-    // ObjectId
+    ObjectId(ObjectType::Type, u32),
```

Comment:
This is a little confusing. It seems like ObjectId always have Type == Device, is this just because everything else is TODO? Any reason why you can't just translate to a nested enum here? e.g.

    enum ObjectType {
        Device(u32),
        SomethingElse(SomeOtherDataType),
    }

then when unmarshalling just something like

    match &body[0] {
        8 => ObjectType::Device( ? ), // Okay I dont know where this comes from
    }